### PR TITLE
FINERACT-2073: Client search v2 using Client Identifiers

### PR DESCRIPTION
**Pull request by alberto-art3ch**
## Description

Client search v2 using Client Identifiers in the search values as we have in the general search endpoint. There is not impact or change needed in the UI

[FINERACT-2073](https://issues.apache.org/jira/browse/FINERACT-2073)

Screenshot 2024-03-26 at 12 50 40 a m

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-core/src/main/java/org/apache/fineract/portfolio/client/domain/Client.java [hunk 1]
```diff
@@ -220,6 +222,9 @@ public class Client extends AbstractAuditableWithUTCDateTimeCustom {
     @Column(name = "proposed_transfer_date")
     private LocalDate proposedTransferDate;
 
+    @OneToMany(cascade = CascadeType.ALL, mappedBy = "client", orphanRemoval = true, fetch = FetchType.LAZY)
+    protected Set<ClientIdentifier> identifiers = new HashSet<>();
```
Comment (adamsaghy) [inline fineract-core/src/main/java/org/apache/fineract/portfolio/client/domain/Client.java:226]: Why do we need this new association? It is not used anywhere

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Kindly add the proper FINERACT story for this PR and check my review!

**Comment by alberto-art3ch** [comment]
> Kindly add the proper FINERACT story for this PR and check my review!

Done

**Comment by adamsaghy** [comment]
Please update the Commit message too!

**Comment by adamsaghy** [comment]
Please update the Commit message too!

**Comment by alberto-art3ch** [comment]
> Please update the Commit message too!

Done! Commit message updated

**Review by adamsaghy — APPROVED** [review]
LGTM

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Please check the failing build!

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**